common/k8s/src/restarting_stream.rs

```diff
@@ -0,0 +1,60 @@
+use std::pin::Pin;
```

Comment:
At some point we should probably split this out into it's own utility crate. It's pretty generic